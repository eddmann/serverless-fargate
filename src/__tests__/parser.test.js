const parse = require('../parser');

test('minimal service task configuration', () => {
  const parsed = parse({
    vpc: {
      securityGroupIds: ['sg-1234'],
      subnetIds: ['subnet-1234'],
    },
    tasks: {
      'my-task': {
        image: 'my-image',
      },
    },
  });

  expect(parsed).toMatchSnapshot();
});

test('minimal scheduled task configuration', () => {
  const parsed = parse({
    vpc: {
      securityGroupIds: ['sg-1234'],
      subnetIds: ['subnet-1234'],
    },
    tasks: {
      'my-task': {
        image: 'my-image',
        schedule: 'rate(1 minute)',
      },
    },
  });

  expect(parsed).toMatchSnapshot();
});

test('full service task configuration', () => {
  const parsed = parse({
    clusterName: 'my-cluster-name',
    containerInsights: true,
    memory: '1GB',
    cpu: 512,
    environment: {
      global: 'variable',
    },
    executionRoleArn: 'arn:aws:iam::123456:role/my-custom-execution-role',
    taskRoleArn: 'arn:aws:iam::123456:role/my-custom-task-role',
    iamRoleStatements: [
      {
        Effect: 'Allow',
        Action: ['resource:*'],
        Resource: '*',
      },
    ],
    iamManagedPolicies: ['arn:aws:iam::aws:policy/my-managed-policy'],
    vpc: {
      securityGroupIds: ['sg-1234'],
      subnetIds: ['subnet-1234'],
      assignPublicIp: false,
    },
    tags: {
      global: 'tag',
    },
    tasks: {
      'task-1': {
        name: 'my-task-1',
        image: 'my-image-1',
        executionRoleArn:
          'arn:aws:iam::123456:role/my-custom-execution-role-for-task-1',
        taskRoleArn: 'arn:aws:iam::123456:role/my-custom-task-role-for-task-1',
        vpc: {
          securityGroupIds: ['sg-5678'],
          subnetIds: ['subnet-5678'],
          assignPublicIp: true,
        },
        command: ['command'],
        entryPoint: ['entrypoint'],
        memory: '2GB',
        cpu: 1024,
        environment: {
          task: 'variable',
        },
        service: {
          desiredCount: 2,
          maximumPercent: 100,
          minimumHealthyPercent: 0,
          spot: true,
        },
        tags: {
          task: 'tag',
        },
      },
      'task-2': {
        image: 'my-image-2',
      },
    },
  });

  expect(parsed).toMatchSnapshot();
});

test('full service task configuration(ARM64)', () => {
  const parsed = parse({
    clusterName: 'my-cluster-name',
    containerInsights: true,
    memory: '1GB',
    cpu: 512,
    architecture: 'ARM64',
    environment: {
      global: 'variable',
    },
    executionRoleArn: 'arn:aws:iam::123456:role/my-custom-execution-role',
    taskRoleArn: 'arn:aws:iam::123456:role/my-custom-task-role',
    iamRoleStatements: [
      {
        Effect: 'Allow',
        Action: ['resource:*'],
        Resource: '*',
      },
    ],
    iamManagedPolicies: ['arn:aws:iam::aws:policy/my-managed-policy'],
    vpc: {
      securityGroupIds: ['sg-1234'],
      subnetIds: ['subnet-1234'],
      assignPublicIp: false,
    },
    tags: {
      global: 'tag',
    },
    tasks: {
      'task-1': {
        name: 'my-task-1',
        image: 'my-image-1',
        executionRoleArn:
          'arn:aws:iam::123456:role/my-custom-execution-role-for-task-1',
        taskRoleArn: 'arn:aws:iam::123456:role/my-custom-task-role-for-task-1',
        vpc: {
          securityGroupIds: ['sg-5678'],
          subnetIds: ['subnet-5678'],
          assignPublicIp: true,
        },
        command: ['command'],
        entryPoint: ['entrypoint'],
        memory: '2GB',
        cpu: 1024,
        environment: {
          task: 'variable',
        },
        service: {
          desiredCount: 2,
          maximumPercent: 100,
          minimumHealthyPercent: 0,
          spot: true,
        },
        tags: {
          task: 'tag',
        },
      },
      'task-2': {
        image: 'my-image-2',
      },
    },
  });

  expect(parsed).toMatchSnapshot();
});
