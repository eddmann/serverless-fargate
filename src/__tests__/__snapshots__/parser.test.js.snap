// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`full service task configuration 1`] = `
Object {
  "clusterName": "MyFargateCluster",
  "cpu": 512,
  "environment": Object {
    "global": "variable",
  },
  "executionRoleArn": "arn:aws:iam::123456:role/my-custom-execution-role",
  "iamManagedPolicies": Array [
    "arn:aws:iam::aws:policy/my-managed-policy",
  ],
  "iamRoleStatements": Array [
    Object {
      "Action": Array [
        "resource:*",
      ],
      "Effect": "Allow",
      "Resource": "*",
    },
  ],
  "logGroupName": undefined,
  "memory": "1GB",
  "tags": Object {
    "global": "tag",
  },
  "taskRoleArn": "arn:aws:iam::123456:role/my-custom-task-role",
  "tasks": Array [
    Object {
      "command": Array [
        "command",
      ],
      "cpu": 1024,
      "entryPoint": Array [
        "entrypoint",
      ],
      "environment": Object {
        "global": "variable",
        "task": "variable",
      },
      "executionRoleArn": "arn:aws:iam::123456:role/my-custom-execution-role-for-task-1",
      "image": "my-image-1",
      "memory": "2GB",
      "name": "my-task-1",
      "service": Object {
        "desiredCount": 2,
        "maximumPercent": 100,
        "minimumHealthyPercent": 0,
        "spot": true,
      },
      "tags": Object {
        "global": "tag",
        "task": "tag",
      },
      "taskRoleArn": "arn:aws:iam::123456:role/my-custom-task-role-for-task-1",
      "vpc": Object {
        "assignPublicIp": true,
        "securityGroupIds": Array [
          "sg-5678",
        ],
        "subnetIds": Array [
          "subnet-5678",
        ],
      },
    },
    Object {
      "command": Array [],
      "cpu": 512,
      "entryPoint": Array [],
      "environment": Object {
        "global": "variable",
      },
      "executionRoleArn": "arn:aws:iam::123456:role/my-custom-execution-role",
      "image": "my-image-2",
      "memory": "1GB",
      "name": "task-2",
      "service": Object {
        "desiredCount": 1,
        "maximumPercent": 200,
        "minimumHealthyPercent": 100,
        "spot": false,
      },
      "tags": Object {
        "global": "tag",
      },
      "taskRoleArn": "arn:aws:iam::123456:role/my-custom-task-role",
      "vpc": Object {
        "assignPublicIp": false,
        "securityGroupIds": Array [
          "sg-1234",
        ],
        "subnetIds": Array [
          "subnet-1234",
        ],
      },
    },
  ],
  "vpc": Object {
    "assignPublicIp": false,
    "securityGroupIds": Array [
      "sg-1234",
    ],
    "subnetIds": Array [
      "subnet-1234",
    ],
  },
}
`;

exports[`minimal scheduled task configuration 1`] = `
Object {
  "clusterName": "MyFargateCluster",
  "cpu": 256,
  "environment": Object {},
  "executionRoleArn": undefined,
  "iamManagedPolicies": Array [],
  "iamRoleStatements": Array [],
  "logGroupName": undefined,
  "memory": "0.5GB",
  "tags": Object {},
  "taskRoleArn": undefined,
  "tasks": Array [
    Object {
      "command": Array [],
      "cpu": 256,
      "entryPoint": Array [],
      "environment": Object {},
      "executionRoleArn": undefined,
      "image": "my-image",
      "memory": "0.5GB",
      "name": "my-task",
      "schedule": "rate(1 minute)",
      "tags": Object {},
      "taskRoleArn": undefined,
      "vpc": Object {
        "assignPublicIp": false,
        "securityGroupIds": Array [
          "sg-1234",
        ],
        "subnetIds": Array [
          "subnet-1234",
        ],
      },
    },
  ],
  "vpc": Object {
    "assignPublicIp": false,
    "securityGroupIds": Array [
      "sg-1234",
    ],
    "subnetIds": Array [
      "subnet-1234",
    ],
  },
}
`;

exports[`minimal service task configuration 1`] = `
Object {
  "clusterName": "MyFargateCluster",
  "cpu": 256,
  "environment": Object {},
  "executionRoleArn": undefined,
  "iamManagedPolicies": Array [],
  "iamRoleStatements": Array [],
  "logGroupName": undefined,
  "memory": "0.5GB",
  "tags": Object {},
  "taskRoleArn": undefined,
  "tasks": Array [
    Object {
      "command": Array [],
      "cpu": 256,
      "entryPoint": Array [],
      "environment": Object {},
      "executionRoleArn": undefined,
      "image": "my-image",
      "memory": "0.5GB",
      "name": "my-task",
      "service": Object {
        "desiredCount": 1,
        "maximumPercent": 200,
        "minimumHealthyPercent": 100,
        "spot": false,
      },
      "tags": Object {},
      "taskRoleArn": undefined,
      "vpc": Object {
        "assignPublicIp": false,
        "securityGroupIds": Array [
          "sg-1234",
        ],
        "subnetIds": Array [
          "subnet-1234",
        ],
      },
    },
  ],
  "vpc": Object {
    "assignPublicIp": false,
    "securityGroupIds": Array [
      "sg-1234",
    ],
    "subnetIds": Array [
      "subnet-1234",
    ],
  },
}
`;
